sgx-isa/src/arch.rs

```diff
@@ -0,0 +1,74 @@
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
```

Comment:
You shouldn't need to have any `cfg` in this file. This file should only be included when needed, and then everything in this file should be defined.